src/neo-vm/ExecutionEngine.cs

```diff
@@ -61,9 +61,14 @@ internal protected set
             }
         }
 
-        public ExecutionEngine()
+        public ExecutionEngine() : this(new ReferenceCounter())
```

Comment:
Is there any chance that one may may provide a nesting access (call X -> call Y -> call Z -> ...) in such a way that it surpasses limits for array counts, etc, in a dangerous way?

Reply:
In fact, I see that this was the previous behavior... the changed one is on the new sharing constructor. Nevermind.